### DISCUSSION TITLE
Wanna know how VITS works?

### DISCUSSION THREAD

**Discussion by p0p4k** [coqui-ai/TTS#2620] [General]
![image](https://github.com/coqui-ai/TTS/assets/8834712/69eed45c-695f-4a36-801c-fbb28767325a)

I have drawn a simple diagram explaining VITS architecture. 
Following VITS, NaturalSpeech-1 uses a very similar architecture (https://arxiv.org/pdf/2205.04421.pdf).
The DurationPredictor of VITS is also a very interesting architecture in its own. If anyone has any questions or wants to go indepth, I can help them.

**Comment by p0p4k** [comment]
For alignment: We first calculate logprobab of z_p of every phoneme for every timestep in the spectrogram using the projections m,_p and logs_p. Then we use MAS to find a path which maximizes the total logprobab of chosen z_p at every timestep. So, now for each timestep we have a single m_p and logs_p. This m_p, logs_p of timestamp length vector is where our z_p is supposed to be sampled from. Thus, the KL loss.

**Reply by Oorgien** [reply]
Hi! Could you please give more details on how does the Duration Predictor works?

**Reply by p0p4k** [reply]
@Oorgien add me on discord `p0p4k`

**Reply by CasonTsai** [reply]
> For alignment: We first calculate logprobab of z_p of every phoneme for every timestep in the spectrogram using the projections m,_p and logs_p. Then we use MAS to find a path which maximizes the total logprobab of chosen z_p at every timestep. So, now for each timestep we have a single m_p and logs_p. This m_p, logs_p of timestamp length vector is where our z_p is supposed to be sampled from. Thus, the KL loss.

so  can we get the phoneme duration time according to output of StochasticDurationPredictor/DurationPredictor in inference?
